client/state/sharing/keyring/actions.js

```diff
@@ -12,9 +17,11 @@ import {
 /**
  * Triggers a network request for a user's connected services.
  *
+ * @param {Function} successCallback Function that is called after a successful
+ *                                   request. Default: _.identity.
  * @return {Function} Action thunk
  */
-export function requestKeyringConnections() {
+export function requestKeyringConnections( successCallback = identity ) {
```

Comment:
In the specific example, how about something like:

```js
popupMonitor.once( 'close', () => {
	// When the user has finished authorizing the connection
	// (or otherwise closed the window), force a refresh
	this.props.requestKeyringConnections();
	this.setState( { isAwaitingPreSelectRequest: true } );
} );
```

Then a `componentWillReceiveProps`:

```js
componentWillReceiveProps( nextProps ) {
	if ( this.props.isRequestingConnections && ! nextProps.isRequestingConnections && 
			this.state.isAwaitingPreSelectRequest ) {
		this.setState( {
			isSelectingAccount: true,
			isAwaitingPreSelectRequest: false
		} );
	}
}
```